js/repl/Repl.js

```diff
@@ -187,14 +189,13 @@ export default class Repl extends React.Component {
     );
   }
 
-  _setupBabel() {
-    loadBabel(this.state.babel, this._workerApi, babelState => {
-      this.setState(babelState);
+  async _setupBabel() {
```

Comment:
`async` / `await` will require us to use the runtime polyfill, right? Which is pretty big.

Reply:
Yeah... I haven't measured the weight of it yet. I could switch it back to regular promises but async/await is so nice 😢 Particularly for code like this where you need to conditionally await something:
```
      if (!isBuildNumeric) {
        // Build in URL is *not* numeric, assume it's a branch name
        // Get the latest build number for this branch
        build = await loadLatestBuildNumberForBranch(
          config.circleciRepo,
          build
        );
      }
      const url = await loadBuildArtifacts(config.circleciRepo, build, doLoad);
```

I figured the developer experience is worth the overhead, but I'll need to see exactly how large the overhead is.

There's an alternative transform that transforms them into generators instead, which is lighter but browser support isn't as good (https://caniuse.com/#feat=es6-generators).

Reply:
Yeah, I agree- the syntax is super nice. But `babel-polyfill.min` is 104.56 kB. That's why the new REPL only lazily loads it if the user selects "Evaluate"

Reply:
I'm not dead set against changing this. It just seemed...questionable.

If we decide it's okay to add that weight, I'll happily replace the other promises with async/await.

Reply:
I'm not loading babel-polyfill, I'm only loading the Regenerator runtime. It looks like that's 6.8 KB gzipped unminified (https://unpkg.com/regenerator-runtime@0.11.0/runtime.js) but I haven't checked the minified size yet.

Reply:
@Daniel15 gzipped is 2.7kb

Reply:
2.7 KB is likely fine given babel itself is over 2 MB 😛  what do you think, @bvaughn?

Reply:
@Daniel15 + babel-preset-env ~1mb 🤕

Reply:
Oh! I didn't realize you could load just the regenerator runtime separately.

That's great then! Let's definitely do that.

Reply:
PS. With regard to `babel-preset-env` size, we're lazy-loading everything that we can in the new REPL. What we can't lazy-load (like Babel standalone and React) we're at least loading from a CDN.

Any ideas to improve this are welcome though!

Reply:
^ btw we should be using useBuiltIns if we are using env so it would remove polyfills that aren't used.

Reply:
Using built-ins is a configurable option ^